### PR TITLE
feat(client/photo): add the ability to use a custom image server

### PR DESCRIPTION
**Pull request by AvarianKnight**
Adds the option for people to use a external image server and config variables to do so.

### CODE REVIEW

**Comment by RockySouthpaw** [comment]
Can you update the read me with this PR?

**Review by TasoOneAsia — APPROVED** [review]
Looks fine for now aside from not cleaning up in some bad areas. But will worry about that later.

**Comment by TasoOneAsia** [comment]
@RockySouthpaw docs are not yet public, so nothing relevant to change yet. Ill probably public them tomorrow.

File: resources/client/cl_photo.ts [hunk 1]
```diff
@@ -84,21 +85,29 @@ onNet(PhotoEvents.SEND_PHOTOS, (photos: string[]) => {
...
+
   exp['screenshot-basic'].requestScreenshotUpload(
-    'https://api.imgur.com/3/image',
-    'imgur',
+    config.images.url,
+    config.images.type,
     {
+      encoding: 'jpg',
       headers: {
         authorization: `Client-ID ${SCREENSHOT_BASIC_TOKEN}`,
```
Comment (TasoOneAsia) [inline resources/client/cl_photo.ts:95]: With the change to add custom image servers, there may be points where an authorization header isn't required. We should handle for that.

File: resources/client/cl_photo.ts [hunk 2]
```diff
@@ -84,21 +85,29 @@ onNet(PhotoEvents.SEND_PHOTOS, (photos: string[]) => {
...
         'content-type': 'multipart/form-data',
       },
     },
     (data: string) => {
       try {
-        const imageLink = JSON.parse(data).data.link;
+        let imageLink: string;
+        if (config.images.isCustomImageServer) {
+         imageLink = `${config.images.url.includes('https') ? 'https' : 'http'}://${JSON.parse(data).files[0].url}`;
```
Comment (TasoOneAsia) [inline resources/client/cl_photo.ts:106]: This can be a variable resp format.

**Review by TasoOneAsia — CHANGES_REQUESTED** [review]
Few things I missed first time around

**Comment by itschip** [comment]
@AvarianKnight Are you able to take a look at the comments from Taso?

**Comment by itschip** [comment]
@AvarianKnight Please switch branch to 'develop', not 'master'

**Comment by TasoOneAsia** [comment]
Revamped by #321

**Pull request closed without merging**